zebra-network/src/peer/client.rs

```diff
@@ -318,13 +348,15 @@ impl Service<Request> for Client {
         // The current task must be scheduled for wakeup every time we return
         // `Poll::Pending`.
         //
-        // `poll_canceled` schedules the client task for wakeup
-        // if the heartbeat task exits and drops the cancel handle.
+        // `check_heartbeat` and `check_connection` schedules the client task for wakeup
+        // if the either task exits.
```

Comment:
Nit: confusing grammar

```suggestion
        // `check_heartbeat` and `check_connection` schedule the client task for wakeup
        // if either task exits, or if the heartbeat task drops the cancel handle.
```

Reply:
Done.